### PR TITLE
Added deeplink to open meeting stage view

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**

### CODE REVIEW

File: samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ const AppInMeeting = props => {
     useEffect(() => {
         microsoftTeams.app.initialize();
         microsoftTeams.app.getContext().then((context) => {
+            console.log("context is: " + JSON.stringify(context));
```
Comment (ChetanSharma-msft) [inline samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx:19]: I think, we can remove it.

File: samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx [hunk 2]
```diff
@@ -33,6 +34,23 @@ const AppInMeeting = props => {
...
     }, []);
 
+    const openDeepLink = () => {
+        microsoftTeams.app.initialize();
+        var appContext = JSON.stringify({
+            "appSharingUrl": `${window.location.origin}/todoView`,
+            "appId": "<<App id>>", "useMeetNow": false
+        });
+
+        alert(appContext);
```
Comment (ChetanSharma-msft) [inline samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx:44]: remove alert

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments

File: samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx [hunk 3]
```diff
@@ -33,6 +34,23 @@ const AppInMeeting = props => {
         });
     }, []);
 
+    const openDeepLink = () => {
```
Comment (ChetanSharma-msft) [inline samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx:37]: Add a comment like deep link to meeting stage view.

File: samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx [hunk 4]
```diff
@@ -33,6 +34,23 @@ const AppInMeeting = props => {
...
+            "appSharingUrl": `${window.location.origin}/todoView`,
+            "appId": "<<App id>>", "useMeetNow": false
+        });
+
+        alert(appContext);
+
+        var encodedContext = encodeURIComponent(appContext).replace(/'/g, "%27").replace(/"/g, "%22");
+
+        var shareToStageLink = `https://teams.microsoft.com/l/meeting-share?deeplinkId={sampleid}&fqdn=teams.microsoft.com&lm=deeplink%22&appContext=${encodedContext}`;
+
```
Comment (ChetanSharma-msft) [inline samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx:49]: Update the readme to add the doc section link in Further reading section.

File: samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx [hunk 5]
```diff
@@ -33,6 +33,21 @@ const AppInMeeting = props => {
         });
     }, []);
 
+    const openDeepLink = () => {
```
Comment (ChetanSharma-msft) [inline samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx:36]: Add a comment like deep link to meeting stage view.

File: samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx [hunk 6]
```diff
@@ -33,6 +33,21 @@ const AppInMeeting = props => {
...
+        microsoftTeams.app.initialize();
+        var appContext = JSON.stringify({
+            "appSharingUrl": `${window.location.origin}/todoView`,
+            "appId": "<<App id>>", "useMeetNow": false
+        });
+
+        var encodedContext = encodeURIComponent(appContext).replace(/'/g, "%27").replace(/"/g, "%22");
+
+        var shareToStageLink = `https://teams.microsoft.com/l/meeting-share?deeplinkId={sampleid}&fqdn=teams.microsoft.com&lm=deeplink%22&appContext=${encodedContext}`;
```
Comment (ChetanSharma-msft) [inline samples/meetings-stage-view/csharp/AppInMeeting/ClientApp/src/components/app-in-meeting.jsx:45]: Update the readme to add the doc section link in Further reading section.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments

File: samples/meetings-stage-view/csharp/README.md [hunk 7]
```diff
@@ -171,3 +171,4 @@ For reference please check [Enable and configure your apps for Teams meetings](h
 ## Further Reading.
 
 [Meeting stage view](https://learn.microsoft.com/en-us/microsoftteams/platform/sbs-meetings-stage-view)
```
Comment (ChetanSharma-msft) [inline samples/meetings-stage-view/csharp/README.md:173]: We need to update the README for app id/external id for deep link to work locally.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my comment

File: samples/meetings-stage-view/csharp/README.md (hunk 7 above)
Comment (Prithvi-MSFT) [reply samples/meetings-stage-view/csharp/README.md:186]: Updated readme

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Review by Wajeed-msft — APPROVED** [review]

**Comment by Prithvi-MSFT** [comment]
**Build Pipeline status** - All builds successful.
image

**Pull request merged**